src/main/java/io/github/debuggyteam/tablesaw/integration/emi/TableSawEmiRecipe.java

```diff
@@ -0,0 +1,82 @@
+package io.github.debuggyteam.tablesaw.integration.emi;
+
+import dev.emi.emi.api.recipe.EmiRecipe;
+import dev.emi.emi.api.recipe.EmiRecipeCategory;
+import dev.emi.emi.api.render.EmiTexture;
+import dev.emi.emi.api.stack.EmiIngredient;
+import dev.emi.emi.api.stack.EmiStack;
+import dev.emi.emi.api.widget.WidgetHolder;
+import io.github.debuggyteam.tablesaw.TableSaw;
+import io.github.debuggyteam.tablesaw.api.TableSawRecipe;
+import net.minecraft.item.ItemStack;
+import net.minecraft.recipe.Ingredient;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class TableSawEmiRecipe implements EmiRecipe {
+    Identifier id;
+    EmiIngredient input;
+    EmiStack output;
```

Comment:
These should probably have visibility modifiers
```suggestion
    private Identifier id;
    private EmiIngredient input;
    private EmiStack output;
```

Reply:
done in latest push